src/parser/mod.rs

```diff
@@ -12289,6 +12352,137 @@ impl<'a> Parser<'a> {
         }
         false
     }
+
+    /// Look for an expected keyword, without consuming it
+    fn peek_keyword(&self, expected: Keyword) -> bool {
+        match self.peek_token().token {
+            Token::Word(w) => expected == w.keyword,
+            _ => false,
+        }
+    }
+
+    /// Look for one of expected keyword, without consuming it
+    fn peek_keywords(&self, expected: &[Keyword]) -> bool {
+        for kw in expected {
+            if self.peek_keyword(*kw) {
+                return true;
+            }
+        }
+        false
+    }
+
+    fn parse_show_opt_in(&mut self) -> Result<Option<ShowStatementIn>, ParserError> {
+        let clause = match self.parse_one_of_keywords(&[Keyword::FROM, Keyword::IN]) {
+            Some(Keyword::FROM) => ShowStatementInClause::FROM,
+            Some(Keyword::IN) => ShowStatementInClause::IN,
+            _ => return Ok(None),
+        };
+
+        if self.parse_keyword(Keyword::DATABASE) {
```

Comment:
the branches look identical, can we do the match once so that we reuse the same lookahead logic for all branches? thinking e.g roughly
```rust
        let (parent_type, parent_name) = match self.parse_one_of_keywords(&[Keyword::DATABASE,])? {
            Some(Keyword::DATABASE) => (Some(ShowStatementInParentType::Database), None)
           // ...
            None => { /* mysql */ }
        };
        if parent_type.is_some() && self.peek_keywords(&[Keyword::STARTS, Keyword::WITH]) {
            Ok(Some(ShowStatementIn {
                clause,
                parent_type,
                parent_name,
            }))
        } else {
            let parent_name = match self.parse_object_name(false) {
                Ok(n) => Some(n),
                Err(_) => None,
            };
            Ok(Some(ShowStatementIn {
                clause,
                parent_type: Some(ShowStatementInParentType::Database),
                parent_name,
            }))
        }
```